lib/cyrusdb_twoskip.c

```diff
@@ -620,15 +622,15 @@ static int read_onerecord(struct dbengine *db, size_t offset,
 
     /* long key */
     if (record->keylen == UINT16_MAX) {
-        base = BASE(db) + offset;
-        record->keylen = ntohll(*((uint64_t *)base));
+        ptr = base + offset;
+        record->keylen = ntohll(*((uint64_t *)ptr));
         offset += 8;
     }
```

Comment:
This is just renaming `base` to `ptr` because `base` was a bad name for something that keeps being moved!  The intent is to keep being incremented as we parse each section, since the records are variable length.